wren-ui/src/apollo/server/resolvers/modelResolver.ts

```diff
@@ -310,6 +311,34 @@ export class ModelResolver {
     return data;
   }
 
+  public async getNativeSql(
+    _root: any,
+    args: { responseId: number },
+    ctx: IContext,
+  ): Promise<string> {
+    const { responseId } = args;
+
+    // If using a sample dataset, native SQL is not supported
+    const project = await ctx.projectService.getCurrentProject();
+    const sampleDataset = project.sampleDataset;
+    if (sampleDataset) {
+      throw new Error(`Doesn't support Native SQL`);
+    }
+
+    // get sql statement of a response
+    const response = await ctx.askingService.getResponse(responseId);
+    if (!response) {
+      throw new Error(`Thread response ${responseId} not found`);
+    }
+
+    // construct cte sql and format it
+    const steps = response.detail.steps;
+    const stepIndex = steps.length - 1;
```

Comment:
You don't have to pass stepIndex in if you only want to construct the Cte SQL of all steps.

I already did it here if you didn't pass in stepIndex
https://github.com/Canner/WrenAI/blob/eaa63ea4ef4edd6b7b7b117ffffef175f77196e9/wren-ui/src/apollo/server/services/askingService.ts#L99

Reply:
Thanks for the reminder.
Done by 11751a63